src/components/RightSidebar/Description/Description.vue

```diff
@@ -0,0 +1,327 @@
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<div class="description"
+		:class="{'description--editing': editing}">
+		<RichContentEditable
+			ref="contenteditable"
+			v-model="descriptionText"
+			class="description__contenteditable"
+			:auto-complete="()=>{}"
+			:maxlength="maxLength"
+			:contenteditable="editing && !loading"
+			:placeholder="placeholder"
+			@submit="handleSubmitDescription"
+			@keydown.esc="handleCancelEditing" />
+		<template v-if="!loading">
+			<template v-if="editing">
+				<button
+					class="description__button"
+					:aria-label="t('spreed','Cancel editing description')"
+					@click="handleCancelEditing">
+					<Close decorative
+						:size="20" />
+				</button>
+				<button
+					class="description__button primary"
+					:aria-label="t('spreed','Submit conversation description')"
+					:disabled="!canSubmit"
+					@click="handleSubmitDescription">
+					<Check decorative
+						:size="20" />
+				</button>
+				<div v-if="showCountDown"
+					v-tooltip.auto="countDownWarningText"
+					class="counter"
+					tabindex="0"
+					aria-label="countDownWarningText">
+					<span>{{ charactersCountDown }}</span>
+				</div>
+			</template>
+			<button v-if="!editing"
+				class="description__button"
+				:aria-label="t('spreed','Edit conversation description')"
+				@click="handleEditDescription">
+				<Pencil decorative
+					:size="20" />
+			</button>
+		</template>
+		<div v-if="loading" class="icon-loading-small spinner" />
+	</div>
+</template>
+
+<script>
+import Pencil from 'vue-material-design-icons/Pencil'
+import Check from 'vue-material-design-icons/Check'
+import Close from 'vue-material-design-icons/Close'
+import RichContentEditable from '@nextcloud/vue/dist/Components/RichContenteditable'
+import Tooltip from '@nextcloud/vue/dist/Directives/Tooltip'
+
+export default {
+	name: 'Description',
+	components: {
+		Pencil,
+		Check,
+		Close,
+		RichContentEditable,
+	},
+
+	directives: {
+		Tooltip,
+	},
+
+	props: {
+		/**
+		 * The description (An editable paragraph just above the sidebar tabs)
+		 */
+		descriptionTitle: {
+			type: String,
+			default: t('spreed', 'Description'),
+		},
+
+		/**
+		 * A paragraph below the title.
+		 */
+		description: {
+			type: String,
+			default: '',
+		},
+
+		/**
+		 * Shows or hides the editing buttons.
+		 */
+		editable: {
```

Comment:
This is not used fully? Currently guests see the edit button. The save button is not clickable thou